mobile/src/navigations/onboarding-navigator/components/steps/steps.tsx

```diff
@@ -13,17 +15,51 @@ import {
     TalentOnboardingScreenNumber,
     TextCategory,
 } from '~/bundles/common/enums/enums';
+import { useAnimatedStyle } from '~/bundles/common/hooks/hooks';
 import { globalStyles } from '~/bundles/common/styles/styles';
 import {
     type TalentOnboardingRouteProperties,
     type ValueOf,
 } from '~/bundles/common/types/types';
 
 import { Step } from '../components';
+import { ANIMATION_VALUES } from './constants/constants';
 import { styles } from './styles';
 
+const {
+    ICON_FINAL_X_COORDINATES,
+    TEXT_FINAL_Y_COORDINATES,
+    ICON_INITIAL_X_COORDINATES,
+    TEXT_INITIAL_Y_COORDINATES,
+    INITIAL_PROGRESS_VALUE,
+    FINAL_PROGRESS_VALUE,
+} = ANIMATION_VALUES;
+
 const Steps: React.FC<DrawerContentComponentProps> = (props) => {
     const { navigation, state } = props;
+    const progress = useDrawerProgress();
+
+    const textAnimatedStyle = useAnimatedStyle(() => {
+        const translateY = interpolate(
+            progress.value,
+            [INITIAL_PROGRESS_VALUE, FINAL_PROGRESS_VALUE],
+            [TEXT_INITIAL_Y_COORDINATES, TEXT_FINAL_Y_COORDINATES],
+        );
+        return {
+            transform: [{ translateY }],
+        };
+    });
+
+    const iconAnimatedStyle = useAnimatedStyle(() => {
+        const translateX = interpolate(
+            progress.value,
+            [INITIAL_PROGRESS_VALUE, FINAL_PROGRESS_VALUE],
+            [ICON_INITIAL_X_COORDINATES, ICON_FINAL_X_COORDINATES],
+        );
+        return {
+            transform: [{ translateX }],
+        };
+    });
```

Comment:
Maybe put it in a separate function, like useAnimationStyles? And then use it for textAnimatedStyle and iconAnimatedStyle and maybe somewhere else later.

Reply:
Guess it's a good idea, if we will add more animations in our project, waiting for Katerina's opinion